packages/manager/apps/catalog/src/pages/index.tsx

```diff
@@ -35,12 +34,15 @@ export default function Catalog() {
 
   useEffect(() => {
     if (products.length > 0) {
-      const searchParams = getSearchUrlFromFilterParams(
+      const customSearchParams = getSearchUrlFromFilterParams(
         searchText,
         categories,
         universes,
       );
-      navigate({ search: searchParams });
+      if (routerInitiated) setSearchParams(customSearchParams);
+      else if (!routerInitiated && customSearchParams.size !== 0) {
```

Comment:
`routerInitiated` is `false` here so no to check it again.
Can you add `{}` to ease the readability as well ?
```suggestion
      else if (customSearchParams.size !== 0) {
```